File: app/src/main/java/com/returnnotfound/stain/data/remote/AnimeService.java
package com.returnnotfound.stain.data.remote;

import com.returnnotfound.stain.data.dto.AnimeTopWraperDTO;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface AnimeService {
  @GET("top/anime/{page}/{subtype}")
  Single<AnimeTopWraperDTO> getTop(@Path("page") int page, @Path("subtype") String subType);
}

File: app/src/main/java/com/returnnotfound/stain/data/remote/AnimeService.kt
package com.returnnotfound.stain.data.remote

import com.returnnotfound.stain.data.dto.AnimeTopWrapperDTO
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path

interface AnimeService {
  @GET("top/anime/{page}/{subtype}")
  fun getTop(
    @Path("page") page: Int,
    @Path("subtype") subType: String?
  ): Single<AnimeTopWrapperDTO?>?
}